lib/src/keyboard.dart

```diff
@@ -81,7 +82,40 @@ extension KeyboardInput on WidgetTester {
 
   Future<void> pressEnter() async {
     await sendKeyEvent(LogicalKeyboardKey.enter, platform: _keyEventPlatform);
-    await pumpAndSettle();
+  }
+
+  /// Simulates the user pressing ENTER on a widget that could be attached to the IME.
+  /// 
+  /// If the ENTER key isn't handled, generates a "\n" insertion followed by a TextInputAction.newline.
```

Comment:
I don't think we want this method to be adaptive, I think we want it to only send to IME. Then we can have a `pressEnterAdaptive()` that uses this conditional check to send one or the other.

I say this because what if the tester truly wants to send through the IME only? Seems like we'd lack a tool for them to do that.

Reply:
Makes sense. Updated.